codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -272,6 +275,13 @@ private boolean readClientPrefaceString(ByteBuf in) throws Http2Exception {
             if (bytesRead == 0 || !ByteBufUtil.equals(in, in.readerIndex(),
                                                       clientPrefaceString, clientPrefaceString.readerIndex(),
                                                       bytesRead)) {
+                String http1Request =
```

Comment:
I don't think your proposed change works.  The "HTTP/1.1" is the last part of the first line, not at the beginning.  ByteBuf lacks a "contains" method, which is why I am using String.

An example request looks like:

```
POST /my/path HTTP/1.1
Host: myserver.com
Authorization: Bearer 12345

```

The code needs to find the `/HTTP/1\.[01]/`in the first line (delimited by an `\r\n` usually, but possibly just `\n`).  If present, it extract all data up to that sequence, which exclude all the subsequent headers.